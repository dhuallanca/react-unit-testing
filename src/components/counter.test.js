import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Counter from './counter';

describe('Pruebas de Contador', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Counter></Counter>);
  })

  test('validar el renderizado con snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  test('validar el incremento del contador', () => {
    wrapper.find('button').first().simulate('click');
    expect(wrapper.find('h1').text()).toEqual('1');
  })

  test('validar el decrement del contador', () => {
    wrapper.find('button').last().simulate('click');
    expect(wrapper.find('h1').text()).toEqual('-1');
  })
});