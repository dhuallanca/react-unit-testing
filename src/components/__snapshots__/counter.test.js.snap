// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de Contador validar el renderizado con snapshot 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Incrementar
  </button>
  <button
    onClick={[Function]}
  >
    Disminuir
  </button>
  <h1>
    0
  </h1>
</div>
`;
